feat/#5 DDD 패턴으로 변경

Goldbar97 opened the pull request:
## 변경 타입
- [ ] 신규 기능 추가/수정
- [ ] 버그 수정
- [x] 리팩토링
- [ ] 설정
- [ ] 비기능 (주석 등 기능에 영향을 주지 않음)

## 변경 내용
- **as-is**
    - Controller, Service, Repository 계층으로 도메인 관리

- **to-be**
    - DDD 패턴으로 변경하기 위해 계층을 4개로 나누고 이동

## 코멘트
- (추가적인 설명이나 코멘트가 필요한 경우 여기에 작성)

iconew123 committed:
#9 feat: delivery-service 모듈 생성

- delivery-service 의 유레카 클라이언트 모듈 생성
- ORM, DB, SWAGGER 의존성 추가

iconew123 committed:
#9 feat: delivery-service 의 설정파일 생성

- application.yml 파일 설정 (환경변수 관리 및 db연결, 포트설정 등)
- NAVER MAP API등록
- DB설정
- 도커에 DB 컨테이너 설정

iconew123 committed:
#9 fix: delivery-service gradle 버그 수정

- dependencyManagement가 dependencies 상위에 있어야
spring cloud dependencies를 불러올 수 있음

iconew123 committed:
#9 feat: delivery-service active 설정

- 로컬환경으로 프로파일 설정

iconew123 committed:
#9 feat: delivery-service 의 JPA 추가

- JPA 설정 추가
- data.sql 미리 만들기

iconew123 committed:
#9 feat: delivery-service 의 JPA 추가

- FeignClient 추가

STW5 committed:
#25 feat: user - JWT 발급 및 검증 로직 추가

STW5 committed:
#25 feat: user - 로그인 DTO 생성

STW5 committed:
#25 feat: user - 로그인 서비스 및 컨트롤러 구현

STW5 committed:
#25 chore: user 모듈 추가 누락 수정

STW5 committed:
#25 refactor: gateway - AuthenticationFilter 불필요한 로그 제거

Goldbar97 committed:
#5 refactor: hub - DDD 구조로 변경

Goldbar97 committed:
#5 refactor: hub - DDD 구조로 변경

Goldbar97 committed:
#5 feat: hub - 네이버 API 연동

Goldbar97 committed:
#5 feat: hub - 네이버 API 연동

Goldbar97 committed:
#5 refactor: hub - DDD 패턴으로 변경

Goldbar97 committed:
#5 refactor: hub - DDD 패턴으로 변경

hanjihae reviewed (commented):
1. presentation - controller, dto
2. application - service, facade
3. domain - entity, repository
4. infrastructure - client나 config 같은 거?
이렇게 구성되는 거 아닌가요...??

iconew123 committed:
#9 feat: delivery-service 허브간 위치 관계 생성 API 구현

- DDD 아키텍쳐로 구성
- FeignClient로 네이버맵 API 호출 (Geocoding, Direction5)
- 1. 허브 주소를 받아 경도 위도 받아오기
- 2. 경,위도 조합으로 허브간 거리관계(M)와 시간(분) 측정

iconew123 committed:
#9 feat: delivery-service Optimizer

- Optimizer

iconew123 committed:
Merge pull request #29 from fifteen-night/feat/#9_createRoute

Feat/#9 루트경로 생성 (허브 이름으로 직접 저장)

iconew123 reviewed (approved):
dto 에 res, req로도 나누면 보기 더 편할거같아요

STW5 committed:
Merge pull request #27 from fifteen-night/feat/#25_userSignIn

feat/#25_로그인 기능 구현

Goldbar97 commented:
> 1. presentation - controller, dto
> 2. application - service, facade
> 3. domain - entity, repository
> 4. infrastructure - client나 config 같은 거?
>    이렇게 구성되는 거 아닌가요...??

dto 는 presentation, application 둘 다 있는데 저는 일단 한쪽에 두기로 했습니다. config 도 infra에 넣을 지 고민했는데 팀장님 참고해서 일단 따로 빼긴 했습니다.

Goldbar97 committed:
#5 refactor: hub - dto 디렉토리 세부화

Goldbar97 committed:
#5 refactor: hub - dto 디렉토리 세부화

Goldbar97 committed:
#5 feat: hub - Querydsl 의존성 및 ignore 추가

Goldbar97 committed:
#5 feat: hub - Mapper 에 toDto 메서드 추가

Goldbar97 committed:
#5 feat: hub - querydsl config 추가

Goldbar97 committed:
#5 feat: hub - Querydsl 검색 기능 추가

Goldbar97 committed:
#5 chore: hub - 충돌 해결

STW5 reviewed (approved):
수고하셨습니다!
